src/safeposix/syscalls/net_calls.rs

```diff
@@ -3080,22 +3080,68 @@ impl Cage {
         }
     }
 
-    // Because socketpair needs to spawn off a helper thread to connect the two ends
-    // of the socket pair, and because that helper thread, along with the main
-    // thread, need to access the cage to call methods (syscalls) of it, and because
-    // rust's threading model states that any reference passed into a thread but
-    // not moved into it mut have a static lifetime, we cannot use a standard member
-    // function to perform this syscall, and must use an arc wrapped cage
-    // instead as a "this" parameter in lieu of self
+    /// ## ------------------SOCKETPAIR SYSCALL------------------
+    /// ### Description
+    /// The `socketpair_syscall()` call creates an unnamed pair of connected sockets
+    /// in the specified domain, of the specified type, and using the optionally
+    /// specified protocol.
+    /// The file descriptors used in referencing the new sockets are returned
+    /// in sv.sock1 and sv.sock2. The two sockets are indistinguishable.
+    /// ### Function Arguments
+    /// The `socketpair_syscall()` receives four arguments:
+    /// * `domain` -  The domain argument specifies a communication domain; this
```

Comment:
I'm not sure if this is wrapping more than 80 chars. Regardless make sure to cargo fmt this before we merge.